src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/MSBuildShellOutNuGetProject.cs

```diff
@@ -326,6 +355,51 @@ public static string GetMSBuildPath(IVsBuildPropertyStorage buildPropertyStorage
             return null;
         }
 
+        public bool IsRestoreRequired(
+            IEnumerable<VersionFolderPathResolver> pathResolvers,
+            ISet<PackageIdentity> packagesChecked,
+            ExternalProjectReferenceContext context)
+        {
+            return true;
+        }
+
+        public async Task<IReadOnlyList<ExternalProjectReference>> GetProjectReferenceClosureAsync(
+            ExternalProjectReferenceContext context)
+        {
+            await TaskScheduler.Default;
```

Comment:
Since `MSBuildShellOutNuGetProject` has `IsRestoreRequired` always returning true, saving the closure to the context will not save any time. The context is passed in because that is part of the interfaces (it is useful in the UWP project.json and packages.config cases).